notify/impl.go

```diff
@@ -62,6 +66,79 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// End-users should not be exposed to Go's type system,
+// as this will confuse them and prevent simple things like
+// simple equality checks to fail. Map everything to float64/string.
+type templateData struct {
+	Status                      string
+	Alerts                      []templateAlert
+	AlertCommonLabels           map[string]string
+	AlertCommonLabelnamesSorted []string
+	GroupLabels                 map[string]string
+	GroupLabelnamesSorted       []string
+}
+
+type templateAlert struct {
+	Labels      map[string]string
+	Annotations map[string]string
+}
+
+func generateTemplateData(ctx context.Context, as ...*types.Alert) *templateData {
```

Comment:
Summing up the above, I'm not sure about this whole mapping here. I would like to see some use cases causing trouble exposing the original types/data structures with their potentially useful methods.